### ISSUE TITLE
[CZ:DS 8265] Game crashes when mission Recoil starts

### ISSUE THREAD

**Issue by CS-PRO1** [ValveSoftware/halflife#2548]
I was testing another issue earlier and came to this as well. Once the helicopter take down scene is done the game crashes. The performance in intro is terrible too
Other missions are working correctly

**Comment by CS-PRO1** [comment]
![Untitled](https://user-images.githubusercontent.com/50755959/60042471-7d37a300-96c6-11e9-8ecf-04b7b53785a9.png)

That's the only fishy thing I got in console

**Comment by wilianmaique** [comment]
fix the problems of _period sounds that never exist.

**Comment by HackZork** [comment]
I can replicate this crash too. It happens only when deleted scenes beta (build 8265) is selected. However, I don't think those sound errors cause the crash, since they appear during the cutscene and the game crashes immediately after the first level load. Here is an excerpt from my qconsole.log: [https://gist.github.com/HackZork/a82627cd7f97965cf6c7943f0ffc4ff5](url)

**Comment by mikela-valve** [comment]
Thanks @CS-PRO1, I was able to reproduce this and tracked it to being caused by the fix applied in #2272. This and possibly a number of other places rely on a bug in `PEntityOfEntIndex` that causes the lookup for the player's entity to fail but then still be resolved due to logic in `CBaseEntity::Instance` that always returns the world entity for NULL. This happens because the player hasn't been fully instantiated at that point in the scene so it ends up crashing rather than erroneously using the world entity in place of the player entity for a bit. 

This will be fixed in the next beta.

**Comment by CS-PRO1** [comment]
Lmao I've never seen a game depending on a bug to work correctly. I'll scout the rest of missions to see if any others have the same issue..

**Comment by mikela-valve** [comment]
This should be "fixed" in beta 8279.

**Comment by CS-PRO1** [comment]
@mikela-valve Great, I'll close this and move further discussion to https://github.com/ValveSoftware/halflife/issues/2272
